### ISSUE TITLE
Error saying mark: {commit_no} not declared

### ISSUE THREAD

**Issue by abhinavchawla** [newren/git-filter-repo#134]
Receiving this error, why would this be happening?
```
Parsed 35296 commits
fatal: mark :35390 not declared
fast-import: dumping crash report to .git/fast_import_crash_72586
Error: fast-import failed; see above.
```

Nothing much in the log file as well.

**Comment by newren** [comment]
Could you provide the command line you are running, and the contents of the file referenced in the error message (.git/fast_import_crash_72586 in this case)?  What version of filter-repo are you running with?  What version of git?  Is the repository you are hitting this error with public?  If not, does it also trigger on some public repository?

**Comment by philipsd6** [comment]
Here's my specific case with a public repo:
```sh
$ git clone https://github.com/ansible/awx.git awx.awx
Cloning into 'awx.awx'...
remote: Enumerating objects: 134, done.
remote: Counting objects: 100% (134/134), done.
remote: Compressing objects: 100% (65/65), done.
remote: Total 255004 (delta 88), reused 98 (delta 69), pack-reused 254870
Receiving objects: 100% (255004/255004), 231.02 MiB | 4.44 MiB/s, done.
Resolving deltas: 100% (196924/196924), done.
$ cd awx.awx/
$ git filter-repo --subdirectory-filter awx_collection/

[fast_import_crash_1644523.txt](https://github.com/newren/git-filter-repo/files/5344695/fast_import_crash_1644523.txt)

**Comment by newren** [comment]
Thanks for the testcase!

This repository has a tag of a tag (in fact, it's a bit more messed up than that -- 2.0.1 is a tag of a tag of a tag of a tag of a tag of a tag of a tag of a commit, assuming I counted right).  Not only should commits be filtered out by the choice to only include awx_collection, but tags should be as well.  Rewriting of parent commits that get pruned works fine, but when the parent is a tag rather than a commit, it apparently fails.  I think I've got a fix on the branch issue-134-nested-tag-and-filtering-fix but I need to add a testcase for it too...and it's getting really late here so I'm going to sleep.  I'll add one later, but in the mean time, could you try out the branch and verify if it works for you?

**Comment by philipsd6** [comment]
Hard to believe only two lines fixed this problem, but yes, I confirm that it corrects the error in the use-case I posted above:

    $ git clone https://github.com/newren/git-filter-repo.git                                                                   
    $ git checkout issue-134-nested-tag-and-filtering-fix
    $ cd /tmp
    $ git clone https://github.com/ansible/awx.git awx.awx
    $ cd awx.awx
    $ ~/devel/git-filter-repo/git-filter-repo --subdirectory-filter awx_collection/
    Parsed 27419 commits
    New history written in 5.12 seconds; now repacking/cleaning...
    Repacking your repo and cleaning out old unneeded objects
    HEAD is now at 7eebc5440 Merge pull request #8306 from thedoubl3j/sanity_ignore
    Enumerating objects: 4673, done.
    Counting objects: 100% (4673/4673), done.
    Delta compression using up to 4 threads
    Compressing objects: 100% (1213/1213), done.
    Writing objects: 100% (4673/4673), done.
    Total 4673 (delta 2603), reused 4643 (delta 2581)
    Completely finished after 5.59 seconds.

**Comment by newren** [comment]
Okay, I added a testcase and pushed it up, so fixed with commit 7eaaf191de (filter-repo: correctly prune nested tags not matching filtering criteria, 2020-10-08).
